tools/rosbag/src/recorder.cpp

```diff
@@ -403,8 +403,16 @@ void Recorder::startWriting() {
 
 void Recorder::stopWriting() {
     ROS_INFO("Closing '%s'.", target_filename_.c_str());
-    bag_.close();
-    rename(write_filename_.c_str(), target_filename_.c_str());
+    try
```

Comment:
introducing try&catch seems to be reasonable.

but I'd do check the callstack where this `Recorder::stopWriting` is called, and then add try&catch statement on the caller of `Recorder::stopWriting` to minimize the influence to other stack.

if i am not mitaken, this changes behavior of `Recorder::checkDuration` and `Recorder::checkSize`. with this fix, they will move on process with error message.

Reply:
I agree, it seems like it would be better to handle this exception somewhere else where we can properly stop the process instead of letting the recorder continue. I presume it doesn't make sense to keep recording if there is an IO exception.